app/src/main/java/com/andrew/tasky/domain/AgendaItems.kt

```diff
@@ -52,4 +52,15 @@ object AgendaItems {
                 agendaItemListSorted.add(it)}}
         return agendaItemListSorted
     }
+
+    fun addAgendaItem(newAgendaItem: AgendaItem){
+        agendaItemList.add(newAgendaItem)
+    }
+
+    fun replaceAgendaItem(newAgendaItem: AgendaItem, oldAgendaItem: AgendaItem){
+        agendaItemList[agendaItemList.indexOf(oldAgendaItem)] = newAgendaItem
+    }
+    fun deleteAgendaItem(agendaItem: AgendaItem){
+        agendaItemList.remove(agendaItem)
```

Comment:
Is there a reason why you have all this in a separate class? In the end this list of agenda items is UI state and should be put in a ViewModel. I'd also recommend using an immutable list instead, since mutable ones are very vulnerable to race conditions